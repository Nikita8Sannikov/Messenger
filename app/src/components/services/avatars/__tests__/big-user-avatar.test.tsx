import { render } from "@testing-library/react";

import BigUserUserAvatar from "@components/services/avatars/big-user-avatar";

describe("BigUserUserAvatar", () => {
	it("should render correctly with all props", () => {
		const props = {
			userId: "123",
			src: "https://example.com/avatar.jpg",
			alt: "User Avatar",
		};

		const { container } = render(<BigUserUserAvatar {...props} />);
		expect(container).toMatchSnapshot();
	});

	it("should render correctly with different props", () => {
		const props = {
			userId: "456",
			src: "https://example.com/different-avatar.jpg",
			alt: "Different User Avatar",
		};

		const { container } = render(<BigUserUserAvatar {...props} />);
		expect(container).toMatchSnapshot();
	});

	it("should render correctly with empty alt text", () => {
		const props = {
			userId: "789",
			src: "https://example.com/avatar.jpg",
			alt: "",
		};

		const { container } = render(<BigUserUserAvatar {...props} />);
		expect(container).toMatchSnapshot();
	});
});
