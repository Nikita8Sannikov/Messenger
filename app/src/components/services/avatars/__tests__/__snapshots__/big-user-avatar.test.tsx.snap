// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigUserUserAvatar should render correctly with all props 1`] = `
<div>
  <span
    class="MuiBadge-root avatar-badge css-63ezxx-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular avatar user-avatar big-avatar css-fmeqll-MuiAvatar-root"
      id="user-avatar"
    >
      <img
        alt="User Avatar"
        class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
        src="https://example.com/avatar.jpg"
      />
    </div>
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-aa1yeu-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`BigUserUserAvatar should render correctly with different props 1`] = `
<div>
  <span
    class="MuiBadge-root avatar-badge css-63ezxx-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular avatar user-avatar big-avatar css-fmeqll-MuiAvatar-root"
      id="user-avatar"
    >
      <img
        alt="Different User Avatar"
        class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
        src="https://example.com/different-avatar.jpg"
      />
    </div>
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-aa1yeu-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`BigUserUserAvatar should render correctly with empty alt text 1`] = `
<div>
  <span
    class="MuiBadge-root avatar-badge css-63ezxx-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular avatar user-avatar big-avatar css-fmeqll-MuiAvatar-root"
      id="user-avatar"
    >
      <img
        alt=""
        class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
        src="https://example.com/avatar.jpg"
      />
    </div>
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-aa1yeu-MuiBadge-badge"
    />
  </span>
</div>
`;
